src/classes/BDI_DataImportService.cls

```diff
@@ -878,6 +878,18 @@ global with sharing class BDI_DataImportService {
                 return false;
             }
 
+            // If the destination field is a Boolean and the DI field intermediate requires interpretation to Boolean
+            // This allows for mapping checkboxes to a tri-state picklist - None, True, False - or a text with either 
+            // blank, "True", or "False"
+            String destinationSObjectname = destinationRecord.getSObjectType().getDescribe().getName(); 
+            if (!(value instanceof Boolean) && 
```

Comment:
I know were nesting a few ifs here, but I might consider adding one more and putting the `destinationSObjectname` var inside the first `value instanceof Boolean` if block, since we don't need to get the SObject name to inspect this unless the field is Boolean.

Reply:
Another option could be to extract the comparison to a little utility method, for readability - something like `if(booleanMappingIsValid(value, destinationSObjectname)`.  This is good - I'm just trying to think through it if it could be a little more readable what with multiple calls to those UTIL classes here.